src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -471,13 +471,6 @@ private bool IsLayoutFixedInCurrentVersionBubbleInternal(TypeDesc type)
 
             if (!NodeFactory.CompilationModuleGroup.VersionsWithModule(defType.Module))
             {
-                if (!type.IsValueType)
-                {
-                    // Eventually, we may respect the non-versionable attribute for reference types too. For now, we are going
-                    // to play it safe and ignore it.
-                    return false;
-                }
-
                 // Valuetypes with non-versionable attribute are candidates for fixed layout. Reject the rest.
                 return type is MetadataType metadataType && metadataType.IsNonVersionable();
```

Comment:
Does this mean that ArrayType and ParameterizedType aren't non-versionable? I was under the impression that in the PR title you mentioned them as non-versionable. In other words, I'm wondering whether it might be easier to have IsNonVersionable accept a TypeDesc and checking the various type classes there. Non-blocking, just curious.